### PR TITLE
feat(stack-view): Remove shim and fallbacks

### PR DESCRIPTION
**Pull request by valentin-mladenov**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [x] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number:   CDE-1224

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

## Other information

BREAKING CHANGE: These custom properties are deprecated or deleted in favor of new styles:
```
| --clr-stack-view-stack-block-border-bottom                             | Deprecated. Use `--clr-stack-view-border-color` instead         |
| --clr-stack-view-stack-children-stack-block-border-bottom-color        | Deprecated. Use `--clr-stack-view-border-color` instead         |
| --clr-stack-view-stack-block-label-font-size                           | Deprecated in favor of `--clr-stack-view-row-font-size`         |
| --clr-stack-view-stack-block-label-font-weight                         | Deprecated in favor of `--clr-stack-view-row-font-weight`       |
| --clr-stack-view-stack-block-label-line-height                         | Deprecated in favor of `--clr-stack-view-row-line-height`       |
| --clr-stack-view-stack-block-label-letter-spacing                      | Deprecated in favor of `--clr-stack-view-row-letter-spacing`    |
| --clr-stack-view-color                                                 | Deprecated in favor of `--clr-stack-view-row-color`             |
| --clr-stack-view-bg-color                                              | Deprecated in favor of `--clr-stack-view-row-bg-color`          |
| --clr-stack-view-stack-children-stack-block-label-and-content-bg-color | Deprecated in favor of `--clr-stack-view-row-bg-color`          |
| --clr-stack-view-stack-block-caret-color                               | Deprecated in favor of `--clr-stack-view-expandable-row-color`  |
| --clr-stack-view-stack-block-expanded-bg-color                         | Deprecated in favor of `--clr-stack-view-expanded-row-bg-color` |
| --clr-stack-view-stack-block-expandable-hover                          | Deprecated in favor of `--clr-stack-view-expandable-row-hover`  |
| --clr-stack-view-stack-block-expanded-text-color                       | Deprecated in favor of `--clr-stack-view-expanded-row-color`    |
| --clr-stack-view-stack-block-expanded-text-color                       | Deprecated in favor of `--clr-stack-view-expanded-row-color`    |
| --clr-stack-view-stack-block-label-text-color                          | DELETED. Replaced with `--clr-stack-view-row-color`             |
| --clr-stack-view-stack-block-label-and-content-bg-color                | DELETED. Replaced with `--clr-stack-view-row-bg-color`          |
| --clr-stack-view-font-size                                             | DELETED. Replaced with `--clr-stack-view-row-font-size`         |
| --clr-stack-view-font-weight                                           | DELETED. Replaced with `--clr-stack-view-row-font-weight`       |
| --clr-stack-view-line-height                                           | DELETED. Replaced with `--clr-stack-view-row-line-height`       |
| --clr-stack-view-letter-spacing                                        | DELETED. Replaced with `--clr-stack-view-row-letter-spacing`    |
```

### CODE REVIEW

File: projects/angular/src/data/stack-view/_stack-view.clarity.scss [hunk 1]
```diff
@@ -199,154 +70,77 @@
...
-      // It's just easier to align and stretch everything vertically this way than with
-      // floats or inline-blocks.
       flex: 1 1 auto;
       display: flex;
+      align-items: center;
 
       // ATM
       // before section is caret placeholder remove it and replace it with pure padding right
-      // padding-right: var(--cds-global-space-9);
+      // padding-right: tokens.$cds-global-space-9;
```
Comment (Jinnie) [inline projects/angular/src/data/stack-view/_stack-view.clarity.scss:109]: Just remove maybe?

File: projects/angular/src/data/stack-view/_stack-view.clarity.scss [hunk 2]
```diff
@@ -199,154 +70,77 @@
       // This way we keep a border-bottom on the last element if the stack view is higher
       // than the total height of the  blocks it contains, but we do not have a
       // weird 1px scroll when both heights match exactly.
-      @include lastElementBottomBorder(variables.$clr-use-custom-properties);
+      border-bottom: none;
     }
 
-    .stack-block-changed {
-      > .stack-block-label {
-        margin-left: -1 * variables.$clr_baselineRem_0_375;
```
Comment (Jinnie) [inline projects/angular/src/data/stack-view/_stack-view.clarity.scss:207]: Can't find what happened with this one?

File: projects/angular/src/data/stack-view/_stack-view.clarity.scss [hunk 3]
```diff
@@ -9,97 +9,32 @@
...
-    box-shadow: $shadow;
-  }
-}
+@use '@cds/core/tokens/tokens.scss';
 
 @include meta.load-css('properties.stack-view');
 
 @include mixins.exports('stack-view.clarity') {
   .stack-header {
-    font-weight: stack-view-variables.$clr-stack-view-font-weight;
```
Comment (Jinnie) [inline projects/angular/src/data/stack-view/_stack-view.clarity.scss:30]: Why do we remove this? 
Is the default good enough? NG theme was different?

File: projects/angular/src/data/stack-view/_stack-view.clarity.scss [hunk 4]
```diff
@@ -199,154 +70,77 @@
...
-      @include mixins.css-var(
-        color,
-        clr-stack-view-stack-block-content-text-color,
-        stack-view-variables.$clr-stack-view-stack-block-content-text-color,
-        variables.$clr-use-custom-properties
-      );
       flex: 1 1 auto;
       width: 60%;
       margin-bottom: 0;
-      font-weight: stack-view-variables.$clr-stack-view-font-weight;
```
Comment (Jinnie) [inline projects/angular/src/data/stack-view/_stack-view.clarity.scss:349]: Same as previous. Were we overriding different NG setting with this?
Why did we need it?

**Review by Jinnie — COMMENTED** [review]
Some minor questions mostly.
Managed to follow most of the bigger refactors and optimisations, but missing a few ones.

File: projects/angular/src/data/stack-view/_stack-view.clarity.scss (hunk 2 above)
Comment (valentin-mladenov) [reply projects/angular/src/data/stack-view/_stack-view.clarity.scss:207]: New visual test is created for it [stack-view_clrSbNotifyChange](https://1071--storybook-clarity-design.netlify.app/?path=/story/stack-view-stack-block--default&args=clrSbNotifyChange:true). 

Originally the stack view row label was pushed right by the triangle in the picture, but now the triangle have `position: absolute;` and the negative margin is not needed any more. 

image

File: projects/angular/src/data/stack-view/_stack-view.clarity.scss (hunk 1 above)
Comment (valentin-mladenov) [reply projects/angular/src/data/stack-view/_stack-view.clarity.scss:109]: I kept this ATM comment for refactoring JIRA investigation, but can remove it altogether.

File: projects/angular/src/data/stack-view/_stack-view.clarity.scss [hunk 5]
```diff
@@ -9,97 +9,32 @@
...
-          // Weird negative margin to make the button aligned with the stack view
-          // in its default state. It then looks unaligned on hover only.
-          @include mixins.css-var(
-            margin-right,
-            clr-stack-view-stack-title-btn-margin-right,
-            -1 * variables.$clr_baselineRem_0_5,
-            variables.$clr-use-custom-properties
-          );
-        }
+        min-width: tokens.$cds-global-space-11;
```
[the remark this answers is not in the document]
Comment (valentin-mladenov) [reply projects/angular/src/data/stack-view/_stack-view.clarity.scss:37]: `min-width` was originally on `.btn` level, but was overriden by other `min-width` rules. By placing it on the `.btn` parent it ensures that Btn text wont be pushed inward by higher `min-width`.

`.btn` min width is 72px.

With min width
image

Without min width
image

[the remark this answers is not in the document]
Comment (Jinnie) [reply projects/angular/src/data/stack-view/_stack-view.clarity.scss:37]: Thanks

File: projects/angular/src/data/stack-view/_stack-view.clarity.scss (hunk 3 above)
Comment (Jinnie) [reply projects/angular/src/data/stack-view/_stack-view.clarity.scss:30]: Thanks for the discussion. I agree it's okay to get this removed and documented in the STYLES file.

**Review by Jinnie — APPROVED** [review]
LGTM

File: projects/angular/src/data/stack-view/_stack-view.clarity.scss (hunk 1 above)
Comment (valentin-mladenov) [reply projects/angular/src/data/stack-view/_stack-view.clarity.scss]: comment removed in commit be6430d

File: projects/angular/src/data/stack-view/_stack-view.clarity.scss (hunk 3 above)
Comment (valentin-mladenov) [reply projects/angular/src/data/stack-view/_stack-view.clarity.scss:30]: Docs added in commit comment removed in commit be6430d

File: projects/angular/src/data/stack-view/_stack-view.clarity.scss (hunk 4 above)
Comment (valentin-mladenov) [reply projects/angular/src/data/stack-view/_stack-view.clarity.scss:349]: Docs added in commit comment removed in commit be6430d

**Pull request merged**